File: backend/src/main/java/com/dcsuibian/entity/Topic.java
package com.dcsuibian.entity;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.Setter;

import java.time.Instant;
import java.util.List;

@Getter
@Setter
public class Topic {
    private Long id;
    private String content;
    private String title;
    private String cover;
    private User author;
    private User editor;
    private Instant time;
    @JsonManagedReference
    private List<TopicComment> comments;
    @Getter(AccessLevel.NONE)
    @Setter(AccessLevel.NONE)
    private String __abstract;

    public void setAbstract(String value) {
        this.__abstract = value;
    }

    public String getAbstract() {
        return __abstract;
    }
}

File: nodejs-spider/src/models/topic.ts
export interface TopicVO {
  id: number | null;
  title: string | null;
  abstract: string | null;
  author: null | {
    id: number | null;
    name: string;
  };
  editor: null | {
    id: number | null;
    name: string;
  };
  time: number | null;
  content: string | null;
  cover: string | null;
}
